pkg/ipamclaimswebhook/podmutator_test.go

```diff
@@ -39,6 +40,7 @@ type testConfig struct {
 	inputNADs                 []*nadv1.NetworkAttachmentDefinition
 	inputPod                  *corev1.Pod
 	expectedAdmissionResponse admission.Response
+	matcherFunc               interface{}
```

Comment:
Let's maek this more explicit and not use a very opaque `interface{}` function, we can split the expectations for the response into two:

```golang

admissionv1 "k8s.io/api/admission/v1"
type type testConfig struct { 
....
   expectedAdmissionResponse dmissionv1.AdmissionResponse
   expectedAdmissionPatches  types.GomegaMatcher
}
```

Then this can be use like 

```golang
		testConfig{
			inputVM:  dummyVM(nadName),
			inputVMI: dummyVMI(nadName),
			inputNADs: []*nadv1.NetworkAttachmentDefinition{
				dummyNAD(nadName),
			},
			inputPod: &corev1.Pod{},
			expectedAdmissionResponse: admissionv1.AdmissionResponse{
					Allowed:   true,
					PatchType: &patchType,
				},
			expectedAdmissionPatches: Equals([]jsonpatch.JsonPatchOperation{
				{
					Operation: "add",
					Path:      "/metadata/annotations/k8s.ovn.org~1primary-udn-ipamclaim",
					Value:     "vm1.podnet",
				},
				{
					Operation: "replace",
					Path:      "/metadata/annotations/k8s.v1.cni.cncf.io~1networks",
					Value:     "[{\"name\":\"supadupanet\",\"namespace\":\"ns1\",\"ipam-claim-reference\":\"vm1.randomnet\"}]",
				},
			}),
		},
```

Reply:
done, ptal